manifests/repo.pp

```diff
@@ -18,23 +18,11 @@
     descr    => "Pulpcore ${version}",
     baseurl  => pick($baseurl, "https://yum.theforeman.org/pulpcore/${version}/${dist_tag}/\$basearch"),
     enabled  => '1',
-    gpgcheck => if $gpgkey == '' { '0' } else { '1' },
-    gpgkey   => pick($gpgkey, "https://yum.theforeman.org/pulpcore/${version}/GPG-RPM-KEY-pulpcore"),
+    gpgcheck => if $gpgkey == '' or $version == 'nightly' { '0' } else { '1' },
+    gpgkey   => if $version == 'nightly' { undef } else { pick($gpgkey, "https://yum.theforeman.org/pulpcore/${version}/GPG-RPM-KEY-pulpcore") },
```

Comment:
I think you don't need to `undef` the key here, setting `gpgcheck` to `0` in the line above should be sufficient.